### PR TITLE
Add linting and formatting check

### LINKED ISSUE #43: GitHub Actions: Lint + Typecheck

### PR DESCRIPTION
**Pull request by TyHil**
Resolves #43 

Add prettier and ESLint.
Add format and lint scripts in `package.json`.
Run scripts on pre-commit (local) and commit (github).
Run prettier and fix ESLint warnings.

### CODE REVIEW

File: .github/workflows/submit.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: "Submit to Web Store"
+name: 'Submit to Web Store'
```
Comment (Jake3231) [inline .github/workflows/submit.yml:1]: Do we need to make this change?

**Review by Jake3231 — APPROVED** [review]
Tentatively approving after a brief review.

Is this just changes made by the linter? @iamwood Would appreciate your input, as I'm not as experienced in linting for this tech stack.

**Comment by Jake3231** [comment]
Also, have we verified that this works when installing the plugin?

File: .github/workflows/submit.yml (hunk 1 above)
Comment (TyHil) [reply .github/workflows/submit.yml:1]: No, I could exclude this file from Prettier

**Comment by TyHil** [comment]
Yup, pretty much all linter changes.

It works for me, yes.

File: .github/workflows/submit.yml (hunk 1 above)
Comment (Jake3231) [reply .github/workflows/submit.yml:1]: I trust your discretion on that.

**Comment by Jake3231** [comment]
Given that all works I think we're good to merge, but another review would be great.

**Comment by TyHil** [comment]
Just swapped this PRs base branch over to `develop` after I mistakenly had it set to `main` so it should be all good to merge now

**Pull request merged**